python/gui/MaaS/static/maas/js/components/confirmDialog.js

```diff
@@ -0,0 +1,108 @@
+class ConfirmDialog {
+    constructor(parentDivName, id, styleClass, onConfirmFunc) {
+        this.parentDivName = parentDivName;
+        this.id = id;
+        this.styleClass = styleClass;
+        this.onConfirmFunc = onConfirmFunc;
+        
+        this.outer_div = null;
+        this.content_div = null;
+    }
+    
+    get parentDiv() {
+        return document.getElementById(this.parentDivName);
+    }
+}
+
+class ConfirmDeleteDatasetDialog extends ConfirmDialog {
+    constructor(dataset_name, parentDivName, id, styleClass, onConfirmFunc) {
+        super(parentDivName, id, styleClass, onConfirmFunc);
+        this.dataset_name = dataset_name;
+        this.buttons_div = null;
+    }
+    
+    _style_outer_div() {
```

Comment:
I think a lot of this styling will go away in the future, but for now it might make sense to leave it until we have the functionality from this PR and from #210. I would like to revisit the css that we are using GUI-wide in the nearish future.